csharp/src/Ice/Communicator.cs

```diff
@@ -154,11 +155,13 @@ public partial interface Communicator : global::System.IDisposable
         /// </summary>
         ///  <param name="name">The object adapter name.
         ///  </param>
+        ///  <param name="serverAuthenticationOptions">The authentication options used by the ssl transport, pass null if
```

Comment:
I also find the doc-comment unclear / misleading.

First, my understanding is IF you pass null, you get whatever is configured through IceSSL properties, which can be nothing/null too.
Then, what happens:
 - if you don't configure SSL and one of your endpoints is secure (presumably throws exception that you be listed in this doc-comment).
 - if you configure SSL but none of your endpoints is secure (presumably we ignore the SSL config--that's the opposite of what we do in IceRPC)

Reply:
> if you don't configure SSL and one of your endpoints is secure (presumably throws exception that you be listed in this doc-comment).

For a server activate throws, from the Acceptor constructor, if there isn't a certificate configured via IceSSL properties and the server authentications are null. The adapter creation doesn't check IceSSL settings.

For a client no configuration is required.

> if you configure SSL but none of your endpoints is secure (presumably we ignore the SSL config--that's the opposite of what we do in IceRPC

Yes if you pass authentication options to non secure adapter, it will ignore the options.